File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
       Queue<String> ck = new LinkedList<>();
        if(cacheSize == 0){
            answer = cities.length * 5;                
        }
        else{
            for(String s : cities){ 
                s = s.toLowerCase();
                if(!ck.isEmpty() && ck.contains(s)){
                    ck.remove(s);
                    ck.add(s);
                    answer += 1;
                }
                else{
                    answer += 5;
                    if(ck.size() >= cacheSize){
                        ck.remove();
                        ck.add(s);
                    }
                    else{
                        ck.add(s);
                    }
                }

            }
        }
        return answer;
    }
}

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    let answer = 0;
    
    const cache = [];
    if (cacheSize === 0) return cities.length * 5;
    
    for (const city of cities) {
        const lower_city = city.toLowerCase();
        
        if (cache.includes(lower_city)) {
            cache.splice(cache.indexOf(lower_city), 1);
            cache.push(lower_city);
            answer += 1;
        }else {
            if (cache.length === cacheSize) {
                cache.shift();
            }
            cache.push(lower_city);
            answer += 5;
        }
    }
    
    return answer;
}